File: memory/knowledge_hive.py
"""ANT AI Knowledge Hive foundation."""

from ant_common import keyword_filter


class KnowledgeHive:
    def __init__(self):
        self.documents = []

    def add_document(self, document):
        self.documents.append(document)

    def search(self, query):
        return keyword_filter(self.documents, query)

File: memory/KnowledgeHive.kt
package memory

/**
 * Shared memory manager for ANT AI agents.
 */
object KnowledgeHive {
    private val memories = mutableListOf<KnowledgeMemory>()

    fun store(memory: KnowledgeMemory) {
        memories.add(memory)
    }

    fun search(query: String): List<KnowledgeMemory> {
        return memories.filter {
            it.content.contains(query, ignoreCase = true) ||
            it.category.contains(query, ignoreCase = true)
        }
    }

    fun all(): List<KnowledgeMemory> {
        return memories.toList()
    }

    fun clear() {
        memories.clear()
    }
}
